File: frontend/packages/lib-shared/models/order.ts
import type { ModuleField } from './common';
import type { FormDesignConfigDetailParams } from '@lib/shared/models/system/module';

export interface SaveOrderParams {
  name: string;
  customerId: string;
  contractId: string;
  amount?: number;
  owner: string;
  moduleFields?: ModuleField[];
  moduleFormConfigDTO?: FormDesignConfigDetailParams;
  number?: string;
  orderNo?: string;
  processOrderNo?: string;
  processor?: string;
  merchandiser?: string;
  status?: string;
  color?: string;
  colorCode?: string;
  composition?: string;
  materialName?: string;
  materialType?: string;
  processTechnology?: string;
  orderTime?: number;
  quantity?: number;
  unit?: string;
  unitPrice?: number;
  currency?: string;
}

export interface UpdateOrderParams extends SaveOrderParams {
  id: string;
}

export interface OrderItem {
  id: string;
  name: string;
  contractName: string;
  contractId: string;
  moduleFields: ModuleField[];
  createUser: string;
  updateUser: string;
  customerId: string;
  owner: string;
  ownerName?: string;
  number: string;
  orderNo?: string;
  processOrderNo?: string;
  processor?: string;
  merchandiser?: string;
  status?: string;
  color?: string;
  colorCode?: string;
  composition?: string;
  materialName?: string;
  materialType?: string;
  processTechnology?: string;
  orderTime?: number;
  quantity?: number;
  unit?: string;
  unitPrice?: number;
  currency?: string;
  stage: string;
  stageName: string;
  organizationId: string;
  customerName: string;
  createUserName: string;
  updateUserName: string;
  departmentId: string;
  departmentName: string;
  createTime: number;
  updateTime: number;
  amount: number;
  inCustomerPool: boolean;
  poolId: string;
  optionMap?: Record<string, any>;
  attachmentMap?: Record<string, any>;
}

export interface OrderSummaryItem {
  orderNo: string;
  processor?: string;
  owner?: string;
  ownerName?: string;
  merchandiser?: string;
  orderTime?: number;
  quantity?: number | string;
  unit?: string;
  amount?: number | string;
  currency?: string;
}

export interface ExternalOrderSyncResult {
  total: number;
  created: number;
  updated: number;
  skipped: number;
  nextMinId?: number;
  hasMore?: boolean;
  configured: boolean;
  warnings: string[];
}

File: backend/crm/src/main/java/cn/cordys/crm/order/domain/Order.java
package cn.cordys.crm.order.domain;

import cn.cordys.common.domain.BaseModel;
import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.persistence.Table;
import lombok.Data;

import java.math.BigDecimal;

@Data
@Table(name = "sales_order")
public class Order extends BaseModel {

    @Schema(description = "客户id")
    private String customerId;

    @Schema(description = "合同id")
    private String contractId;

    @Schema(description = "联系专员")
    private String owner;

    @Schema(description = "金额")
    private BigDecimal amount;

    @Schema(description = "订单号")
    private String orderNo;

    @Schema(description = "加工单号")
    private String processOrderNo;

    @Schema(description = "加工商")
    private String processor;

    @Schema(description = "跟单员")
    private String merchandiser;

    @Schema(description = "状态")
    private String status;

    @Schema(description = "颜色")
    private String color;

    @Schema(description = "色号")
    private String colorCode;

    @Schema(description = "成分")
    private String composition;

    @Schema(description = "原料名称")
    private String materialName;

    @Schema(description = "原料类型")
    private String materialType;

    @Schema(description = "加工工艺")
    private String processTechnology;

    @Schema(description = "下单时间")
    private Long orderTime;

    @Schema(description = "仓库实际发货日期")
    private Long warehouseActualShipDate;

    @Schema(description = "数量")
    private BigDecimal quantity;

    @Schema(description = "单位")
    private String unit;

    @Schema(description = "单价")
    private BigDecimal unitPrice;

    @Schema(description = "币种")
    private String currency;

    @Schema(description = "组织id")
    private String organizationId;
}
